### PR TITLE
Updating RSPEC texts and rule types

### PR DESCRIPTION
**Pull request by valhristov**

### CODE REVIEW

File: sonaranalyzer-dotnet/build/rspec.ps1 [hunk 1]
```diff
@@ -164,6 +164,7 @@ function CreateStringResources() {
             $description = $description -replace '&amp;', '&'
             $description = $description -replace '&lt;', '<'
             $description = $description -replace '&gt;', '>'
+            $description = $description -replace '\\!', '!'
```
Comment (michalb-sonar) [inline sonaranalyzer-dotnet/build/rspec.ps1]: What is that change about?

File: sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/PackagingTests/DocGeneratorTest.cs [hunk 2]
```diff
@@ -41,7 +41,7 @@ public void CheckNumberOfCrossReferences()
         {
             // These are the deprecated rules which reference their successors
             var crossReferenceCount = GetNumberOfCrossReferences(AnalyzerLanguage.CSharp);
-            crossReferenceCount.Should().Be(6);
+            crossReferenceCount.Should().Be(7);
```
Comment (michalb-sonar) [inline sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/PackagingTests/DocGeneratorTest.cs]: What is that change about?

File: sonaranalyzer-dotnet/build/rspec.ps1 (hunk 1 above)
Comment (valhristov) [inline sonaranalyzer-dotnet/build/rspec.ps1]: That's a workaround for an error in resgen.exe that fails when it sees `\!`... That's present in only one rule and I am not sure if I will not break something else if I fix it...

File: sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/PackagingTests/DocGeneratorTest.cs (hunk 2 above)
Comment (valhristov) [inline sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/PackagingTests/DocGeneratorTest.cs]: This test counts the cross-references between rules. We recently added a reference (e.g. `see` section in the description) in one of the IDisposable rules...

**Pull request merged**